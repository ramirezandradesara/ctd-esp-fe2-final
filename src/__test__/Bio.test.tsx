import { act, fireEvent, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { customRender } from '../test-utils'
import Bio from "../features/bio/Bio";

describe("Pruebas en <Bio />", () => {

    let renderComponent = () => customRender(<Bio />)

    test("Renderizado inicial con el nombre y la biografía de Bart Simpson", async () => {
        renderComponent();

        const bioBart = "A los diez años, Bart es el hijo mayor y único varón de Homero y Marge, y el hermano de Lisa y Maggie. Los rasgos de carácter más prominentes y populares de Bart son su picardía, rebeldía y falta de respeto a la autoridad.";

        expect(screen.getByText('Bart Simpson')).toBeInTheDocument();
        expect(screen.getByText(bioBart)).toBeInTheDocument();
    });


    test("Renderizar nombre y biografía de Homero Simpson", async () => {
        renderComponent();
        const buttonHomero = screen.getByText('HOMERO')
        const bioHomero = "Como jefe de la familia, Homero y su esposa Marge tienen tres hijos: Bart, Lisa y Maggie. Homero trabaja en la planta de energía nuclear de Springfield como inspector de seguridad. Homero encarna muchos estereotipos de la clase trabajadora estadounidense: es obeso, inmaduro, franco, agresivo, calvo, perezoso, ignorante, poco profesional y adicto a la cerveza, a la comida chatarra y a la televisión";

        fireEvent.click(buttonHomero)

        await waitFor(() => expect(screen.queryByText("Homero Simpson")));
        await waitFor(() => expect(screen.queryByText(bioHomero)));
    });

    test("Snapshot de <Bio />", async () => {
        let { asFragment } = customRender(<Bio />)
        let fragment = asFragment()

        expect(fragment).toMatchSnapshot();
    });
});
