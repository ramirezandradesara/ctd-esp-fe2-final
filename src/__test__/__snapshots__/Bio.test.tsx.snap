// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Cita /> Snapshot de <Noticias /> 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe cIkhMV"
  >
    <div
      class="sc-dkrFOg gkXnjm"
    >
      <button
        class="sc-jSUZER cwGpdH"
      >
        BART
      </button>
      <button
        class="sc-jSUZER evzLtq"
      >
        HOMERO
      </button>
      <button
        class="sc-jSUZER evzLtq"
      >
        LISA
      </button>
      <button
        class="sc-jSUZER evzLtq"
      >
        MAGGIE
      </button>
      <button
        class="sc-jSUZER evzLtq"
      >
        MARGE
      </button>
    </div>
    <div>
      <div>
        <img
          alt="Bart Simpson"
          class="sc-gswNZR ctOybF"
          src="bart.png"
        />
      </div>
      <div>
        <h3
          class="sc-hLBbgP dTlAVl"
        >
          Bart Simpson
        </h3>
        <p
          class="sc-eDvSVe gzgjHl"
        >
          A los diez años, Bart es el hijo mayor y único varón de Homero y Marge,
    y el hermano de Lisa y Maggie.
    Los rasgos de carácter más prominentes y populares de Bart son su picardía,
    rebeldía y falta de respeto a la autoridad.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
